tests/test_storage.py

```diff
@@ -0,0 +1,235 @@
+                                            'level', 'year', 'time'])
+
+
+# A function for adding storage technologies and parameterization
+def add_storage_data(scen, time_order):
+    # Adding level of storage
+    scen.add_set('level', 'storage')
+
+    # Adding storage technologies (reservoir, charger, and discharger)
+    scen.add_set('technology', ['dam', 'pump', 'turbine'])
+
+    # Adding a storage commodity
+    scen.add_set('commodity', 'stored_com')
+
+    # Specifying storage reservoir technology
+    scen.add_set('storage_tec', 'dam')
+
+    # Specifying storage level
+    scen.add_set('level_storage', 'storage')
+
+    # Adding mapping for storage and charger/discharger technologies
+    for tec in ['pump', 'turbine']:
+        scen.add_set('map_tec_storage', [tec, 'dam'])
+
+    # Adding time sequence
+    for h in time_order.keys():
+        scen.add_par('time_seq', h, time_order[h], '-')
+
+    # Adding minimum and maximum bound, and losses for storage (percentage)
+    for y in set(scen.set('year')):
+        for h in time_order.keys():
+            scen.add_par('bound_storage_lo', ['node', 'dam', 'electr',
+                                              'storage', y, h], 0.0, '%')
+
+            scen.add_par('bound_storage_up', ['node', 'dam', 'electr',
+                                              'storage', y, h], 1.0, '%')
+
+            scen.add_par('storage_loss', ['node', 'dam', 'electr',
+                                          'storage', y, h], 0.05, '%')
+
+    # input, output, and capacity factor for storage technologies
+    output = {'dam': ['node', 'stored_com', 'storage'],
+              'pump': ['node', 'stored_com', 'storage'],
+              'turbine': ['node', 'electr', 'level']}
+
+    inp = {'dam': ['node', 'stored_com', 'storage'],
+           'pump': ['node', 'electr', 'level'],
+           'turbine': ['node', 'stored_com', 'storage']}
+
+    var_cost = {'dam': 0, 'pump': 0.2, 'turbine': 0.3}
+
+    for y in set(scen.set('year')):
+        for h in time_order.keys():
+            for t in ['dam', 'pump', 'turbine']:
+                tec_sp = ['node', t, y, y, 'mode']
+                scen.add_par('output', tec_sp + output[t] + [h, h], 1, 'GWa')
+                scen.add_par('input', tec_sp + inp[t] + [h, h], 1, 'GWa')
+                scen.add_par('var_cost', tec_sp + [h], var_cost[t], 'USD/GWa')
+
+
+# Main function for building a model and adding seasonality and storage
+def test_storage(test_mp):
+    # First building a simple model
+    scen = Scenario(test_mp, 'no_storage', 'standard', version='new')
+    model_setup(scen, [2020])
+
+    # Initiating missing sets/parameters if not yet in ixmp backend
+    if not storage_in_java:
+        init_storage(scen)
+    scen.commit('initialized test model')
+    scen.solve(case='no_seasonality')
+
+    # Second, adding seasonality
+    seasons_dict = {'a': 0.25, 'b': 0.25, 'c': 0.25, 'd': 0.25}
```

Comment:
these are evenly valued - maybe it is worth testing non-even valued season durations?

Reply:
Thanks @Matt for the suggestion, it's indeed needed! I added another test for inequal time steps.